proxy/round_tripper/proxy_round_tripper.go

```diff
@@ -441,6 +441,11 @@ func isIdempotent(request *http.Request) bool {
 }
 
 func (rt *roundTripper) isRetriable(request *http.Request, err error, trace *requestTracer) (bool, error) {
+	// if the context has been cancelled we do not perform further retries
+	if request.Context().Err() != nil {
+		return false, request.Context().Err()
```

Comment:
As an alternative we could also return the error that has been passed in or combine the two errors to not loose any information:

```suggestion
		return false, fmt.Errorf("%w (%w)", err, request.Context().Err())
```

I'm open for suggestions.

Reply:
i like this suggestion. let's wrap the error to provide all the info.

Reply:
I changed it, but swapped the errors since `context cancelled` is the main reason we are stopping the processing of the request.